project_event/models/project_task.py

```diff
@@ -592,45 +593,41 @@ def reserve_equipment_inside(self, event_id):
 
     @api.multi
     def get_equipment_ids_inside(self):
-        room_id = self.env['resource.calendar.room'].browse(self.room_id).id
+        room_id = self.env['resource.calendar.room'].\
+            browse(self.room_id).id
         return room_id.instruments_ids.ids
 
     @api.multi
     def cancel_resources_reservation(self):
         self.ensure_one()
         if self.reservation_event_id:
-            reservation_event = self.env['calendar.event'].browse(
-                self.reservation_event_id)
-        reservation_event.write(
-            {
-                'state': 'cancelled'
-            }
-        )
+            reserve_event = self.info_calendar_event()
+            reserve_event.write(
+                {'state': 'cancelled'}
+            )
 
     @api.multi
     def draft_resources_reservation(self):
         self.ensure_one()
         if not self.reservation_event_id:
             self.request_reservation()
-        reservation_event = self.env['calendar.event'].browse(
-            self.reservation_event_id)
-        reservation_event.write(
-            {
-                'state': 'draft'
-            }
-        )
+            reserve_event = self.info_calendar_event()
+            reserve_event.write(
+                {'state': 'draft'}
+            )
 
     @api.multi
     def open_resources_reservation(self):
```

Comment:
Looks good, you can improve it more by merging those three methods into one. "def change_resource_reservation(self, status_change)"

Reply:
@eilst  Change according to your comment. Thx you